crates/extensions/c8y_mapper_ext/src/actor.rs

```diff
@@ -111,20 +123,28 @@ impl Actor for C8yMapperActor {
             .send(SyncStart::new(SYNC_WINDOW, ()))
             .await?;
 
-        while let Some(event) = self.messages.recv().await {
-            match event {
-                C8yMapperInput::MqttMessage(message) => {
-                    self.process_mqtt_message(message).await?;
-                }
-                C8yMapperInput::FsWatchEvent(event) => {
-                    self.process_file_watch_event(event).await?;
-                }
-                C8yMapperInput::SyncComplete(_) => {
-                    self.process_sync_timeout().await?;
-                }
-                C8yMapperInput::PublishMessage(message) => {
+        loop {
+            select! {
+                // in some tests may not be connected to the sender, so we need to ignore `None`
+                // here, but should get picked up by the main message box anyway
+                Some(message) = self.publish_messages.recv() => {
                     self.mqtt_publisher.send(message.0).await?;
                 }
+                event = self.messages.recv() => {
+                    let Some(event) = event else { break; };
```

Comment:
The following seems more appropriate: i.e. let consume the published messages till the end even if there is no more input messages.

```suggestion
                Some(event) = self.messages.recv() => {
```

Reply:
No, I'm wrong this would block the mapper on `^C`, as `self.messages.recv()` returns `None` in such a case.